### ISSUE TITLE
Plans for: FreeBSD:13:latest -	10.10.1 ?

### ISSUE THREAD

**Issue by make-nz** [Thefrank/jellyfin-server-freebsd#89]
Last week I sent a mail to Baptiste the maintainer of JF and he answered:

> I have added it to the FreeBSD ports tree so it will be packaged for 13 as it is a supported platform

So, it has landed in _**latest**_ and I upgraded JF in my jail yesterday.
It went through no problems, updated plugins no problems, played music from my flac cd collection no problems, DLNA v5 worked (my issues with it were the same as in v2-3)…
But then bad things got started, 

- no way it would add new files/CDs in rescan
- could not upload images for a folder, artist, album…
- got permission errors when trying to add a new test lib

![grafik](https://github.com/user-attachments/assets/f597a0e8-1254-4918-9fe5-223370132583)

Then I saw this, every time I hit okay and it failed the system created a new folder, adding 2,3,4,5,6
![grafik](https://github.com/user-attachments/assets/0d3c47f2-43c7-402e-a532-fd8b7a241397)

Over night no task was triggered and nothing changed, so **I rolled my jail back**!

@Thefrank do you have plans for 10.10.1 ? I'm happy to try out anything!

**Comment by Thefrank** [comment]
The only significant change between 10.9 and 10.10, for the port, was dropping `sass-embedded` for `sass` but this only impacts the jellyfin-web side and not the server itself

> got permission errors when trying to add a new test lib

All of the issues sound like permission issues. Double check that Jellyfin is at least a member of a group that has read access to the files you want Jellyfin to see. If you are doing library management, which it sounds like you are, then you need to make sure it has write access too. (I am assuming you are using standard permissions and not something complex like ACLs)

**Comment by make-nz** [comment]
I had no permissions issues for years (I also delete files from within JF for years), all other upgrades of JF never caused any permission issues, as I have user/group jellyfin(there once was a change from user jellyfinserver, I remember that I think). 
And after roll back also no permission issues… 
My data is in a data set on it's own pool and mouted to /media/xxx in the JF jail.

And most of all, JF only needs to read the files, all meta data is stored here:
![grafik](https://github.com/user-attachments/assets/5d53a23b-071f-4d60-9d38-ac76477a1538)

JF had no problem with any of the exiting data, reading, displaying or playing in browser, JF App, FinAmp, DLNA on Marantz Stereo.

It beats meet what was going on there.

**Comment by make-nz** [comment]
> not something complex like ACLs

Well no, not complex just grown with every media server I was using since 2014 
Group _Plex_ holds users with full permissions, as I allways liked managing files within the apps.
(got plenty periodic snapshots, but never needed them due to some fatal mistake or app problem)
![grafik](https://github.com/user-attachments/assets/e8369992-1b19-48c2-8048-8a6fed374dc7)


EDIT: Just double checked deleting single files and entire albums still work. Did not change much on the server for month anyway.

**Comment by Thefrank** [comment]
So far, I have not been able to reproduce this behavior on either 13.3 (I still need to update the jailbase) or 14.1.

**Comment by Thefrank** [comment]
@make-nz can you post a log snippet of the errors? 

I did see one difference between mine and ports. I use **SkiaSharp** v2.88.8 and the port uses **Skia** v2.88.3.

The current release of jellyfin wants **SkiaSharp** v2.88.8: https://github.com/jellyfin/jellyfin/blob/f6f4cdf9e788ac522ca6d43eac4570c1fa607da4/Directory.Packages.props#L69 with `main` wanting v2.88.9

**Comment by nx6** [comment]
Hello. I see @make-nz cannot "add new files" -- so I assume we mean scanning and picking up new media on the server? I ask as I am seeing an issue where new video media is not being added to the server. Also, when I try running a manual scan jellyfin is seemingly hanging and not progressing.

I rolled back my jail to 10.9.11 and was able to do a manual scan and "catch up" things fine. Then stopped the server, made a new snapshot, and upgraded again. After the update to 10.10.1 was up I tried a manual scan again, and it got to 68% of all libraries and then stopped. Seeing in console:

`Nov 16 01:47:44 mittsume kernel: pid 40566 (jellyfin), jid 11, uid 868: exited on signal 11`

and looking over the last few days, more occurrences of this error. Maybe the same error is happening when trying to add new music here?

**Comment by nx6** [comment]
I decided to try adding  new libraries, too. Tried a music video library, and then a normal music library. Getting a permissions error on both:

![jf_error](https://github.com/user-attachments/assets/dc8ce336-71da-413c-a31e-b6304f7e9905)

I have checked and the folders have correct permissions.

Meanwhile, in to console I'm seeing the sig 11 exit corresponding to the two attempts to create the new libraries:

Nov 16 02:05:49 mittsume kernel: pid 40669 (jellyfin), jid 11, uid 868: exited on signal 11
Nov 16 02:07:47 mittsume kernel: pid 41203 (jellyfin), jid 11, uid 868: exited on signal 11

**Comment by Thefrank** [comment]
@nx6 This should not resolve the problem...but can you try and replace the current `libskiasharp.so` with the copy that I have? https://github.com/Thefrank/jellyfin-server-freebsd/releases/tag/v10.9.11

Stop/Restart the jellyfin service or restart the jail once you have done that.

Looks like this is showing up on bugs too: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=282704

**Comment by make-nz** [comment]
@nx6 this is pretty much same problem I had, regardless of the file type Vid/Music. Also, have you tried to add a new test lib?
Resulting in this:
![grafik](https://github.com/user-attachments/assets/460784b0-148a-4de5-9bf7-3ec4c6b648ff)


And no, I did not save any log from before rolling back

**Comment by nx6** [comment]
> @nx6 This should not resolve the problem...but can you try and replace the current `libskiasharp.so` with the copy that I have?

~~Where is the existing file located? I can't find it.~~
Nevermind, I found it.

@make-nz 
Yes, I have folders for the test libraries, so it's going through the initial stages of making the libraries, it's just stopping because it can't run an initial scan it seems.
![directory](https://github.com/user-attachments/assets/081334f1-dc27-4a8e-b1b5-47fa4c7d23db)

**Comment by nx6** [comment]
@Thefrank Some odd behavior.

I shut down the server process and deleted the `libskiasharp.so` shortcut and the libskiasharp.so file with the version number at the end I think it was referencing, and copied yours in. After the server came back online I tried a scan of all files and the server picked up a couple things I'd added this morning. I had to kill the tasks because I'd forgotten about the test libraries from last night and they were now appearing, and I think it was trying to catalog all my music files. But I wasn't seeing any exit messages on the console. 👍 

After I have started things back up and I am seeing the permissions error again trying to add another test library. Also scans of all libraries now are just blipping like they aren't really running. But no exit message. I'm gonna roll this back to my snapshot from last night and play with this more. It was a mistake to try and make a "test" music library from my real music files since it would take so long to complete scanning and metadata fetching.

**Comment by make-nz** [comment]
> It was a mistake to try and make a "test" music library from my real music files since it would take so long to complete scanning and metadata fetching.

Just a hint, I took CDs I had never ripped before, to make sure it would not be in any other lib and the detection would trigger my Metadata downloaders set in the test lib.

**Comment by nx6** [comment]
Well it made sense at the time because I do not have any music libraries in my current Jellyfin server, but I had one in a past install so I knew the files already had the correct permissions to be read.

Sticking with the music videos seems better as that's only a handful of files.

**Comment by Thefrank** [comment]
@nx6 
Stop the jellyfin service before making changes. It usually should not be a big issue but this is just to make sure nothing odd happens.
You can `find / -name "libSkiaSharp*"`to make sure all symlinks are removed when you delete the current libSkiaSharp from pkg
Make sure these are removed:
```
/usr/local/lib/libSkiaSharp.so.118.0.0
/usr/local/lib/libSkiaSharp.so
```
This is the file to download: https://github.com/Thefrank/jellyfin-server-freebsd/releases/download/v10.9.11/libSkiaSharp.so
Place it into the same directory where the Jellyfin binary is located, not wherever the libSkiaSharp pkg installs.
Restart Jellyfin or the jail.
If things are still broken: stop jellyfin, delete the libSkiaSharp.so in the Jellyfin directory, and `pkg install -f libskiasharp` to reinstall the library

**Comment by nx6** [comment]
Okay, tried again. Same thing I did before but this time putting `libskiasharp.so` into the jellyfin folder where the executable is, instead of /usr/local/lib/ where I found the other files before. I see the change here. When I looked for the files in 10.9.11 it was just the one file, but 10.10 changes to the pkg install of it and we get the other files.

Anyway, after starting the server back up, tried a scan of all libraries. That took a little time, but it seemed to complete successfully. Then tried making the Music Videos library to test that -- it seems to have worked too. No signs of any Jellyfin crashes/restarts either. So it _did_ resolve the problem. 🥇

**Comment by Thefrank** [comment]
@nx6 @make-nz @bapt
So....
The error seems to come from FreeBSD port version of Skia using `skia_use_system_libjpeg_turbo=true` while I built SkiaSharp with `skia_use_system_libjpeg_turbo=false` 

The root cause is more complicated: 
https://github.com/jellyfin/jellyfin/issues/13016
https://github.com/libjpeg-turbo/libjpeg-turbo/issues/795

edit: cant type when sleepy

**Comment by bapt** [comment]
thanks for the heads up!

**Comment by make-nz** [comment]
> @nx6 @make-nz @bapt So.... The error seems to come from FreeBSD port version of Skia using `skia_use_system_libjpeg_turbo=true` while I built SkiaSharp with `skia_use_system_libjpeg_turbo=false`
> 
> The root cause is more complicated: [jellyfin/jellyfin#13016](https://github.com/jellyfin/jellyfin/issues/13016) [libjpeg-turbo/libjpeg-turbo#795](https://github.com/libjpeg-turbo/libjpeg-turbo/issues/795)
> 
> edit: cant type when sleepy

Glad you and all others helped to narrow it down.
Thanks all

**Comment by robn** [comment]
Updates from elsewhere, just to make sure everyone knows (there's like five separate bug reports open now lol).

jpeg-turbo maintainer believes that Skia is using the API wrong: https://github.com/libjpeg-turbo/libjpeg-turbo/issues/795#issuecomment-2484148592

Skia's "external" jpeg-turbo is 2.1. The offending API change in jpeg-turbo happened in 3.x. I've logged a bug against Skia: https://issues.skia.org/issues/379669745 suggesting they either change their use, or refuse to link against 3.x, or force the use of their bundled version.

I assume from all this that there's no way Skia will be working properly against jpeg-turbo 3.x in the near future, so I expect the only option for Jellyfin for now is to stick to 2.x (by whatever means).

**Comment by Thefrank** [comment]
@robn 
mono/Skia and mono/SkiaSharp do not use `skia_use_system_libjpeg_turbo=true` for any in-tree builds on any platform; which might explain why this has gone uncaught until recently.

mono/Skia was painless to build until it was made downstream of SkiaSharp

My only memories of SkiaSharp are of annoyances in either getting it to build or the hacky workarounds I had to make. 
2/10. 
Would not recommend.

---

**[rant, skip-able]**
Part 1: First look
Skia is downstream of SkiaSharp
SkiaSharp uses .NET and Cake to build
Cake does not know about FreeBSD (This was painless but took a while)
All of Cake's dependencies don't know about FreeBSD (PRs planned where possible)

Part 2: Down an even deeper rabbit hole
Instead of being submodules, SkiaSharp uses Google's tools to pull down a bunch of other repos
Skia is technically a Google project
Google's tools don't know about FreeBSD
Google does not accept PRs for OSes that is does not run in-house (Has this changed?)

Part 3: Actually building
https://github.com/mono/SkiaSharp/blob/v2.88.8/native/linux/build.cake -> Convert to FreeBSD stuff.
With fancy new wheel that has been reinvented it builds!
Tests seem to only cover the C# wrapper working with Skia and not Skia itself.
/tableflip
/quit

Part 4: Fin
Published anyways
Jellyfin tests all passed! 
Jellyfin tests don't fail if Skia version mismatches what Jellyfin wants: add manual check in azure pipeline
**[/rant]**

**Comment by bapt** [comment]
> @make-nz can you post a log snippet of the errors?
> 
> I did see one difference between mine and ports. I use **SkiaSharp** v2.88.8 and the port uses **Skia** v2.88.3.
> 
> The current release of jellyfin wants **SkiaSharp** v2.88.8: https://github.com/jellyfin/jellyfin/blob/f6f4cdf9e788ac522ca6d43eac4570c1fa607da4/Directory.Packages.props#L69 with `main` wanting v2.88.9

Am I missing something ? my understanding is SkiaSharp 2.88.8 is the dotnoet version, I do use it in the port, but the libskiasharp.so itself is a submodule of it which is **Ska** branch **xamarin-mobile-bindings** which hasn't been updated for a year, I haven't done an analysis of the difference between 2.88.3 and the xamarin-mobile-bindings branch yet.

**Comment by Thefrank** [comment]
@bapt You are not missing anything. The whole system is rather confusing!
TL;DR: As long as milestone ~= 118 everything should be fine no matter where you build Skia. The native library is easiest to build from mono/Skia. Building from mono/SkiaSharp is academic (and painful!) unless you want a NuGet for FreeBSD.

Long:
The version check really only matters if Skia < 2.88.0 which is derived from: https://github.com/mono/skia/blob/34086126f05c838a152297fd9c903d80cb11b52f/include/core/SkMilestone.h#L8 on SkiaSharp branch or https://github.com/mono/skia/blob/781c823c12ea98c4d3df7018684cf3ba934bb4e5/include/core/SkMilestone.h#L8 on xamarin-mobile-bindings

SkiaSharp project uses it here: https://github.com/mono/SkiaSharp/blob/main/binding/SkiaSharp/SkiaSharpVersion.cs

Jellyfin?(bindings?) has been more picky: https://github.com/Thefrank/jellyfin-server-freebsd/issues/28 and wants a very constrained range

To make things more confusing: MILESTONE is technically an API version not library version despite the SONAME being used.
Finally: Google's Skia is on 133 so everything is ~1yr behind.

**Comment by nx6** [comment]
10.10.3 is out now, and I tried updating to see what happens, so you don't have to!

- libSkiaSharp has been changed to chromium's libjpeg-turbo version.

The update installed and I deleted thefrank's libSkiaSharp.so from the jellyfin directory before bringing the server back online. The good news is, this version let me run a library scan fine, so that look okay. The bad news is the Server Dashboard is no longer loading. Just Server > Dashboard, the other options are fine. I did try refreshing the page and logging out and back in again. I ran my test scan by going to the Libraries screen and choosing a library that I knew had new content that wasn't scanned in yet.

Just wanted to let folks know here since it resolves this specific issue (though adds a new one).

**Comment by Thefrank** [comment]
See if `pkg install -f libskiasharp` resolves the issue

**Comment by nx6** [comment]
> See if `pkg install -f libskiasharp` resolves the issue

No change there. It got updated to 2.88.3_1 when I updated jellyfin so I think this was already done. Tried refresh and a restart of the entire jail jellyfin is running in as well this time.

**Comment by nx6** [comment]
I removed `libSkiaSharp.so.118.0.0` and the symlink `libSkiaSharp.so` again from /usr/local/lib/ and reinstalled your version in the jellyfin directory to try it out and it's the same, no Admin Dashboard. So seems to be an unrelated issue. There were changes to dotnet in this update, too.

**Comment by Thefrank** [comment]
Can you post logs? Jellyfin is built as Self-Contained so it does not need to have dotnet installed.

**Comment by nx6** [comment]
Sorry, I didn't mean that as an external dependency. I just mentioned it as it was a change listed for the port.

> lang/dotnet: Update to 9.0.0
> Move .NET 8 to lang/dotnet8 

Only a few packages on my system were upgraded here.

```
Nov 22 19:57:10 jellyfin-3 pkg[44485]: curl upgraded: 8.10.1 -> 8.10.1_1 
Nov 22 19:57:10 jellyfin-3 pkg[44485]: aom upgraded: 3.10.0_1 -> 3.11.0 
Nov 22 19:57:10 jellyfin-3 pkg[44485]: libskiasharp upgraded: 2.88.3 -> 2.88.3_1 
Nov 22 19:57:17 jellyfin-3 pkg[44485]: jellyfin upgraded: 10.10.1 -> 10.10.3 
```

I haven't done the cofig file creation the jellyfin site talks about to enable debug logging, and if there are any other (normal automatic) logs I don't know where to find them.

**Comment by Thefrank** [comment]
`/var/db/jellyfin` by default unless you changed `jellyfin_data_dir` in the rc file

**Comment by nx6** [comment]
The logs for today aren't showing much of anything for my attempts to load the Dashboard page for the last few minutes. Just some errors reading a few posters.

```
[2024-11-22 21:51:35.286 -06:00] [INF] [39] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "192.168.9.4" request
[2024-11-22 21:51:35.766 -06:00] [ERR] [38] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/mnt/evens_media/Automated_Movies/The Color Purple (1985)/folder.jpg": InvalidInput
[2024-11-22 21:51:35.766 -06:00] [ERR] [38] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/ec/ec36ac591abb30ba9e8131f47e42dab0/poster.jpg": InvalidInput
[2024-11-22 21:51:35.767 -06:00] [ERR] [38] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/c8/c8acc356da4df51e82a95a30d3eb16bb/poster.jpg": InvalidInput
[2024-11-22 21:51:35.767 -06:00] [ERR] [38] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/92/92f0ae4efa4d42bc193d8535f810cfa4/poster.jpg": InvalidInput
[2024-11-22 21:51:35.874 -06:00] [ERR] [44] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/f0/f0c2fe9b37afaced3c10122095459f46/poster.jpg": InvalidInput
[2024-11-22 21:53:26.905 -06:00] [ERR] [17] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/mnt/evens_media/Automated_Movies/The Color Purple (1985)/folder.jpg": InvalidInput
[2024-11-22 21:53:26.905 -06:00] [ERR] [17] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/ec/ec36ac591abb30ba9e8131f47e42dab0/poster.jpg": InvalidInput
[2024-11-22 21:53:26.905 -06:00] [ERR] [17] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/c8/c8acc356da4df51e82a95a30d3eb16bb/poster.jpg": InvalidInput
[2024-11-22 21:53:26.905 -06:00] [ERR] [17] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/92/92f0ae4efa4d42bc193d8535f810cfa4/poster.jpg": InvalidInput
[2024-11-22 21:53:26.978 -06:00] [ERR] [16] Jellyfin.Drawing.Skia.SkiaEncoder: Unable to determine image dimensions for "/var/db/jellyfin/metadata/library/f0/f0c2fe9b37afaced3c10122095459f46/poster.jpg": InvalidInput
[2024-11-22 21:55:31.778 -06:00] [INF] [11] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "192.168.9.4" closed
[2024-11-22 21:55:32.354 -06:00] [INF] [19] Emby.Server.Implementations.HttpServer.WebSocketManager: WS "192.168.9.4" request
```

**Comment by Thefrank** [comment]
Nothing helpful there. 
Those image dimension errors should be fine unless posters are not rendering.

I would say turn on debug logging but it creates a massive log file very quickly https://jellyfin.org/docs/general/administration/troubleshooting/#debug-logging if you want to give it a shot, go for it.

**Comment by make-nz** [comment]
Just to let you know for now — upgraded to 10.10.3 just now and JF it's not running smooth.
@ nx6 what daoe it show for web versin in you install, please?
![grafik](https://github.com/user-attachments/assets/483bf166-3b7e-4d6f-9ddc-d9523c707eb2)

Things on the dashboard are not loading for me either. Have restarted jail 3rd time now thinkinge it's the plugins updating, but no luck.

**Comment by make-nz** [comment]
> attempts to load the Dashboard page for the last few minutes

Same here now, no dashboard at all anymore
![grafik](https://github.com/user-attachments/assets/0ef5af40-34d7-4b2f-b369-186f7415cb39)

I fixed this, by deleting the lib and a rescan was triggered
`Could not find a part of the path '/media/share/JF-Tests/001'.`
but it did not change the dashboard issue, which I think might be an issue of not being updated properly in the upgrade.
[ERR-Log.txt](https://github.com/user-attachments/files/17879112/ERR-Log.txt)

**Comment by nx6** [comment]
@make-nz 
You need to just refresh your webpage to reload to the current web version. Your browser is trying to reuse a cached copy of the web interface.

**Comment by make-nz** [comment]
Thanks nx6 done this, as allways. Same on incognito mode on mobile, or tablet.
![grafik](https://github.com/user-attachments/assets/ca0d229d-b27d-4f7c-a53e-4dd6a7b395a0)

**Comment by nx6** [comment]
Yeah, I was replying to your first message about the web version when you posted again. Good to see you get that worked out. Almost every time I update Jellyfin the web interface tries to open from the previous version and I have to refresh to sync it up. 

I have nothing to add regarding the Dashboard issue. Haven't had a chance to try enabling debug logging to look at it further.

**Comment by make-nz** [comment]
I got DEBUG running, but nothing super strange, just yet

Andwhen I look at Firefox debugging there is a 404, each time I try loading the Dashboard
> GET
http://192.168.178.134:8096/web/32947.1d95d64c35f97caac382.chunk.js
[HTTP/1.1 404 Not Found 10ms]

Uncaught (in promise) ChunkLoadError: Loading chunk 32947 failed.
(error: http://192.168.178.134:8096/web/32947.1d95d64c35f97caac382.chunk.js)
    j http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    e http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    e http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    i http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    f http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    s http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    s http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    u http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    u http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    f http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    A http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    s http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    s http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    u http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    u http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    A http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    promise callback*55219/t.A/< http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    ri http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    Ss http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    is http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    Bl http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    ws http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    ws http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    ls http://192.168.178.134:8096/web/node_modules.react-dom.bundle.js?3d945e1fe71c5c36ed81:2
    A http://192.168.178.134:8096/web/37869.bundle.js?3d945e1fe71c5c36ed81:2
    C http://192.168.178.134:8096/web/37869.bundle.js?3d945e1fe71c5c36ed81:2
    <anonymous> http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    M http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    P http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    17007 http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    s http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    78786 http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    s http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    90076 http://192.168.178.134:8096/web/node_modules.core-js.bundle.js?3d945e1fe71c5c36ed81:1
    s http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    24639 http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    s http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    <anonymous> http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    O http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    <anonymous> http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
    a http://192.168.178.134:8096/web/runtime.bundle.js?3d945e1fe71c5c36ed81:1
    <anonymous> http://192.168.178.134:8096/web/main.jellyfin.bundle.js?3d945e1fe71c5c36ed81:2
runtime.bundle.js:1:48505

Same on Vivaldi Privat Session:
![grafik](https://github.com/user-attachments/assets/8cb30bc7-105f-4697-bc67-d612501ad079)



I'll let it run and see if they go away:
![grafik](https://github.com/user-attachments/assets/16396589-da70-49c1-8f14-3bae74ab2014)

On Firefox new Privat session and debugger set to NO CACHE
![grafik](https://github.com/user-attachments/assets/65acef7d-d132-4c7e-aaca-4be9aa4034ce)

**Comment by rucko24** [comment]
> @nx6 Stop the jellyfin service before making changes. It usually should not be a big issue but this is just to make sure nothing odd happens. You can `find / -name "libSkiaSharp*"`to make sure all symlinks are removed when you delete the current libSkiaSharp from pkg Make sure these are removed:
> 
> ```
> /usr/local/lib/libSkiaSharp.so.118.0.0
> /usr/local/lib/libSkiaSharp.so
> ```
> 
> This is the file to download: https://github.com/Thefrank/jellyfin-server-freebsd/releases/download/v10.9.11/libSkiaSharp.so Place it into the same directory where the Jellyfin binary is located, not wherever the libSkiaSharp pkg installs. Restart Jellyfin or the jail. If things are still broken: stop jellyfin, delete the libSkiaSharp.so in the Jellyfin directory, and `pkg install -f libskiasharp` to reinstall the library

It has worked correctly with your steps, the last one I didn't need to do the one of `pkg install -f libskiasharp`

![image](https://github.com/user-attachments/assets/09dbbade-1fbf-4de8-8a27-f6a063e2d0b4)


Also my dashboard does not display I will try the steps of others.

**Comment by Thefrank** [comment]
> there is a 404

It sounds like something broke between the upgrade between the versions. You can try and do this:
- stop the jellyfin service
- remove the `jellyfin-web` directory
- force reinstall the `jellyfin` package: `pkg install -f jellyfin`
- restart jellyfin service

if this resolves the issue then I need to figure out how and why the upgrade did not work.

**Comment by make-nz** [comment]
Thanks @Thefrank 
Done what you suggested, but things are just the same
![grafik](https://github.com/user-attachments/assets/773e0189-8f56-42a8-8092-dc94ab42ee62)

This is no Browser Cache issue (did 15 years web dev work) and again from different browser the same 404s
![grafik](https://github.com/user-attachments/assets/a683f52d-0d1f-4eb1-a56a-283e56403c14)

**Comment by make-nz** [comment]
On the up side JF is working well, so far no other issue, other Dashboard and Skia errors on reading images.

BTW, I got a bit of a fleet for stesting:
![grafik](https://github.com/user-attachments/assets/ceaf6fcc-5ad7-4226-8519-14b14e90d240)

**Comment by make-nz** [comment]
Wow, this is crazy! On this old app (**Jellyfin Media Player 1.5.0**) it loads the Dashboard!
![grafik](https://github.com/user-attachments/assets/a9104e77-ffae-45e5-9b5e-f6d3641c1147)

**Comment by make-nz** [comment]
> **Jellyfin Media Player 1.5.0**

From the JF forum I got this answered and it explains the behaviour:

> 

> Older Jellyfin Media Players included their own copy of Jellyfin-web, while versions starting with 1.10.1 used the server's jellyfin-web by default.

**Comment by Thefrank** [comment]
![image](https://github.com/user-attachments/assets/d058c108-7921-44ea-93d4-6125855aa458)
Guess I am building from ports to reinstall this into a new jail to troubleshoot

edit:
![image](https://github.com/user-attachments/assets/66c26360-7fd8-487c-9dcb-489d8a98cf65)
yeap. not working. I blame nodejs. I will look into it.

**Comment by make-nz** [comment]
Sorry, fleels a bit like I'm dragging you into this the 2nd time and I can't help more.
So thanks @Thefrank

**Comment by bapt** [comment]
I can reproduce the empty dashboard as well!

**Comment by bapt** [comment]
should be fixed with 10.10.3_1 sorry about the breakage

**Comment by make-nz** [comment]
> fixed with 10.10.3_1

EDIT: just adding the repo here for future ref.
https://github.com/freebsd/freebsd-ports/tree/3e47fee1f3e6528e14f0d0192e41f2d5e1610e28/multimedia/jellyfin
Thanks so far

**Comment by make-nz** [comment]
> > fixed with 10.10.3_1
> 
> EDIT: just adding the repo here for future ref. https://github.com/freebsd/freebsd-ports/tree/3e47fee1f3e6528e14f0d0192e41f2d5e1610e28/multimedia/jellyfin Thanks so far

The pkg has landed for 13.3 yesterday and indeed has resolved the missing dashboard issue.
Thanks @bapt

**Comment by Thefrank** [comment]
Closing this out as everything has been resolved!
